### PR TITLE
Remove SpannableStringBuilder

### LINKED ISSUE #1978: SpannableStringBuilder.equals() shadow implementation is not complete
robolectric:3.0 SDK 21

A result of the `SpannableStringBuilder.equals()` is differs from the result on a real device if the spannable string has any spans attached to it.

The following test incorrectly fails with roboelectric.

    @Test
    public void testSpannableString() throws Exception {
        SpannableStringBuilder spannableString = new SpannableStringBuilder("test");
        spannableString.setSpan(new UnderlineSpan(), 0, spannableString.length(), 0);
        assertFalse(spannableString.equals("test"));
    }

### PR DESCRIPTION
**Pull request by flyingtoaster0**
Resolves #1978

The overridden implementation for equals in SpannableStringBuilder only compares the resulting String, but not the underlying Spans. This is fixed by using Android's implementation instead.

### CODE REVIEW

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowSpannableStringBuilder.java [hunk 1]
```diff
@@ -12,10 +12,6 @@
 public class ShadowSpannableStringBuilder {
   @RealObject SpannableStringBuilder realSpannableStringBuilder;
 
-  @Implementation @Override public boolean equals(Object obj) {
-    return obj != null && realSpannableStringBuilder.toString().equals(obj.toString());
-  }
-
   @Implementation @Override public int hashCode() {
```
Comment (jongerrish) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowSpannableStringBuilder.java]: You should probably remove hashCode() also

Comment (jongerrish) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowSpannableStringBuilder.java]: Um, actually, you should probably delete the whole shadow then.

Comment (flyingtoaster0) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowSpannableStringBuilder.java]: Yeah, I was just going to suggest that actually. Good call.

**Comment by jongerrish** [comment]
Can you merge your changes?

**Pull request merged**